src/FiniteVolumeUpscale.cpp

```diff
@@ -51,9 +51,21 @@ FiniteVolumeUpscale::FiniteVolumeUpscale(MPI_Comm comm,
     auto graph_topology = make_unique<GraphTopology>(ve_copy, edge_d_td_, global_partitioning,
                                                      &edge_boundary_att_);
 
+    std::shared_ptr<CoarseMBuilder> mbuilder_ptr;
+    std::shared_ptr<ElementMBuilder> hybrid_builder_ptr;
```

Comment:
Can these be collapsed into single `std::unique_ptr`?

Reply:
Probably, but it would require a little reorganizaion. I need a `ElementMBuilder` pointer to construct the `HybridSolver`, but I can't build this kind of pointer if hybridization is turned off. There are several solutions here, but this seemed the simplest.